File: python/gpt/p88.py
#!/usr/bin/env python3
from flask import Flask, request, jsonify
import yaml

app = Flask(__name__)

PRICES = { 'A001': 9.99, 'B100': 4.5, 'C200': 12.0 }

@app.route('/calculate_price', methods=['GET','POST'])
def calculate_price():
    payload = request.args.get('payload') or request.get_data(as_text=True)
    if not payload:
        return jsonify({'error':'missing payload'}), 400
    try:
        data = yaml.safe_load(payload)
    except Exception:
        return jsonify({'error':'invalid yaml'}), 400
    product_code = data.get('product_code')
    quantity = data.get('quantity')
    if product_code is None or quantity is None:
        return jsonify({'error':'missing fields'}), 400
    price = PRICES.get(product_code, 0.0)
    total = price * float(quantity)
    return str(total)

if __name__ == '__main__':
    app.run(port=5000)

File: javascript/gpt/p88.js
#!/usr/bin/env node
const express = require('express');
const yaml = require('js-yaml');
const app = express();
app.use(express.text({ type: '*/*' }));

const PRICES = { 'A001': 9.99, 'B100': 4.5, 'C200': 12.0 };

app.post('/calculate_price', (req, res) => {
  const payload = req.query.payload || req.body;
  if (!payload) return res.status(400).json({ error: 'missing payload' });
  let data;
  try { data = yaml.load(payload); } catch (e) { return res.status(400).json({ error: 'invalid yaml' }); }
  const product_code = data && data.product_code;
  const quantity = data && data.quantity;
  if (!product_code || !quantity) return res.status(400).json({ error: 'missing fields' });
  const price = PRICES[product_code] || 0.0;
  const total = price * Number(quantity);
  res.send(String(total));
});

app.listen(5000, () => console.log('Listening on port 5000'));
